paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -0,0 +1,1175 @@
+    }
+
+    finishLoading(repository)
+  }
+
+  protected fun loadResourceFile(
+    file: PathString,
+    repository: T,
+    shouldParseResourceIds: Boolean
+  ) {
+    val folderName = file.parentFileName
+    if (folderName != null) {
+      val folderInfo = FolderInfo.create(folderName, folderConfigCache)
+      if (folderInfo != null) {
+        val configuration = getConfiguration(repository, folderInfo.configuration)
+        loadResourceFile(file, folderInfo, configuration, shouldParseResourceIds)
+      }
+    }
+  }
+
+  protected open fun finishLoading(repository: T) = processAttrsAndStyleables()
+
+  /**
+   * A hook for loading resource IDs from a R.txt file. This implementation does nothing but subclasses may override.
+   *
+   * @return true if the IDs were successfully loaded from R.txt
+   */
+  protected open fun loadIdsFromRTxt() = false
+
+  override fun isIgnored(
+    fileOrDirectory: Path,
+    attrs: BasicFileAttributes
+  ) = if (fileOrDirectory == resourceDirectoryOrFile) {
+    false
+  } else {
+    fileFilter.isIgnored(fileOrDirectory.toString(), attrs.isDirectory)
+  }
+
+  protected open fun loadPublicResourceNames() {
+    // todo load public resources
+  }
+
+  private fun findResourceFiles(filesOrFolders: List<Path>): List<PathString> {
+    val fileCollector = ResourceFileCollector(this)
+    for (file in filesOrFolders) {
+      try {
+        CancellableFileIo.walkFileTree(file, fileCollector)
+      } catch (e: IOException) {
+        // All IOExceptions are logged by ResourceFileCollector.
+      }
+    }
+    for (e in fileCollector.ioErrors) {
+      LOG.severe("Failed to load resources from $resourceDirectoryOrFile: $e")
+    }
+    fileCollector.resourceFiles.sort() // Make sure that the files are in canonical order.
+    return fileCollector.resourceFiles
+  }
+
+  protected fun getConfiguration(
+    repository: T,
+    folderConfiguration: FolderConfiguration
+  ): RepositoryConfiguration {
+    var repositoryConfiguration = configCache[folderConfiguration]
+    if (repositoryConfiguration != null) {
+      return repositoryConfiguration
+    }
+
+    repositoryConfiguration = RepositoryConfiguration(repository, folderConfiguration!!)
+    configCache[folderConfiguration] = repositoryConfiguration
+    return repositoryConfiguration
+  }
+
+  private fun loadResourceFile(
+    file: PathString,
+    folderInfo: FolderInfo,
+    configuration: RepositoryConfiguration,
+    shouldParseResourceIds: Boolean
+  ) {
+    val folderType = getFolderType(file)
+    if (folderType == VALUES) {
+      if (isXmlFile(file)) {
+        parseValueResourceFile(file, configuration)
+      }
+    } else {
+      if (isXmlFile(file) && FolderTypeRelationship.isIdGeneratingFolderType(folderType)) {
+        parseIdGeneratingResourceFile(file, configuration)
+      }
+
+      parseFileResourceFile(file, folderType, configuration)
+    }
+  }
```

Comment:
this implementation is different from https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:resource-repository/main/java/com/android/resources/base/RepositoryLoader.java;l=465?q=RepositoryLoader

is that something to be addressed in a follow-up?

Reply:
yes, will migrate to same implementation when we switch to BasicResourceItem from AS. For now, our implementation is different. But yes, it will be same eventually